client/ayon_core/hosts/maya/api/lib.py

```diff
@@ -1832,6 +1832,29 @@ def apply_attributes(attributes, nodes_by_id):
                 set_attribute(attr, value, node)
 
 
+def is_valid_reference_node(reference_node):
+    """Return whether Maya considers the reference node a valid reference.
+
+    Maya might report an error when using `maya.cmds.referenceQuery`:
+    Reference node 'reference_node' is not associated with a reference file.
+
+    Note that this does *not* check whether the reference node points to an
+    existing file. Instead it only returns whether maya considers it valid
+    and thus is not an unassociated reference node
+
+    Arguments:
+         reference_node (str): Reference node name
+
+    Returns:
+        bool: Whether reference node is a valid reference
+
+    """
+    sel = OpenMaya.MSelectionList()
+    sel.add(reference_node)
+    depend_node = sel.getDependNode(0)
+    return OpenMaya.MFnReference(depend_node).isValidReference()
```

Comment:
I doubt this code works (I tested this function with some node as input)
it turned out that `getDependNode` expects another argument of type `MObject` 
This [link](http://evgeniyzaitsev.com/2015/11/16/get-all-extra-attributes-maya-python-api/) helped me.

```python
# This one works

ref = "boxRN1"

sel = OpenMaya.MSelectionList()
sel.add(ref)

m_node = OpenMaya.MObject()
sel.getDependNode(0, m_node)

m_refrence = OpenMaya.MFnReference(m_node)
print(m_refrence.isLoaded())
print(m_refrence.isValidReference())
```

Reply:
You are looking at the Maya Python API 1.0 -> `import maya.OpenMaya`
My code is for the Maya Python API 2.0 -> `import maya.api.OpenMaya`

Reply:
Yes. (I have maya 2022 py3.7)
```
# Error: TypeError: file <maya console> line 5: MSelectionList_getDependNode expected 3 arguments, got 2 # 
```

Reply:
Thanks for the error  - I'll look into it. It sounds like you're still using the older Python API of Maya.

This will fail:
```python
from maya import OpenMaya
reference_node = "robo_roboangry_modeling_v001RN"
sel = OpenMaya.MSelectionList()
sel.add(reference_node)
depend_node = sel.getDependNode(0)
print( OpenMaya.MFnReference(depend_node).isValidReference())
```

This should work:
```python
from maya.api import OpenMaya
reference_node = "robo_roboangry_modeling_v001RN"
sel = OpenMaya.MSelectionList()
sel.add(reference_node)
depend_node = sel.getDependNode(0)
print( OpenMaya.MFnReference(depend_node).isValidReference())
```

Reply:
I tested the code above in both maya 2023.3 and maya 2024. both work with from `maya.api import OpenMaya`.
Guess we can implement the backward compatibility for maya 2022.
Like:
```
#Backward Compatibility for importing OpenMaya Library
maya_version = int(cmds.about(version=True))
if maya_version >= 2023:
    from maya.api import OpenMaya
else:
    from maya import OpenMaya
```

Reply:
There's no need for backwards compatibility. The logic works also in those versions. The `maya.api.OpenMaya` library has already been present a long time as far as I know.